plugins/inventory/windows_azure.py

```diff
@@ -121,28 +153,36 @@ def read_settings(self):
 
         # Cache related
         if config.has_option('azure', 'cache_path'):
-            cache_path = config.get('azure', 'cache_path')
-            self.cache_path_cache = cache_path + "/ansible-azure.cache"
-            self.cache_path_index = cache_path + "/ansible-azure.index"
+            cache_path = os.path.expanduser(config.get('azure', 'cache_path'))
+            self.cache_path_cache = cache_path + '/ansible-azure.cache'
+            self.cache_path_index = cache_path + '/ansible-azure.index'
```

Comment:
Note: idiomatically we'd do os.path.join() here instead of putting "/" in explicitly but you don't need to change it if you don't want to as it appears that it didn't trip anyone up previously.